node/netprio_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package node
+
+import (
+	"testing"
+
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+)
+
+// TestBase64AllocboundSize tests that the base64 encoded size of the Nonce is the same as the allocbound
+// used on the Nonce field in the struct for netprio response messages.
+func TestBase64AllocboundSize(t *testing.T) {
```

Comment:
This test if passing when I set `NetPrioResponseTagMaxSize = 1`. 
Should we have a test that fails on the code state before the fix?

Reply:
That is tested by `TestMaxSizesCorrect`. 

To confirm that this test works try reducing `netPrioChallengeSize`.

Reply:
So this test is not checking the bug being fixed here?

Reply:
this PR fixes the `netPrioResponse` decoding issue `prioScheme.VerifyPrioResponse from ::47300: msgp: length overflow: 44 \u003e 32`. This test checks exactly this.

Reply:
@algonautshant to check whether this works set the `netPrioChallengeSize` to 30 or lower (if you set it to 31 it will still get padded to 44) and then run the test.